lib/ansible/modules/cloud/spotinst/spotinst_aws_elastigroup.py

```diff
@@ -0,0 +1,1509 @@
+            has_changed = False
+    else:
+        eg = expand_elastigroup(module, is_update=True)
+
+        if state == 'present':
+            group = client.update_elastigroup(group_update=eg, group_id=group_id)
+            message = 'Updated group successfully.'
+
+            try:
+                roll_config = module.params.get('roll_config')
+                if roll_config:
+                    eg_roll = spotinst.aws_elastigroup.Roll(
+                        batch_size_percentage=roll_config.get('batch_size_percentage'),
+                        grace_period=roll_config.get('grace_period'),
+                        health_check_type=roll_config.get('health_check_type')
+                    )
+                    roll_response = client.roll_group(group_roll=eg_roll, group_id=group_id)
+                    message = 'Updated and started rolling the group successfully.'
+            except SpotinstClientException as exc:
+                message = 'Updated group successfully, but failed to perform roll. Error:' + str(exc)
+            has_changed = True
+
+        elif state == 'absent':
+            try:
+                client.delete_elastigroup(group_id=group_id)
+            except SpotinstClientException as exc:
+                if "GROUP_DOESNT_EXIST" in exc.message:
+                    pass
+                else:
+                    module.fail_json(msg="Error while attempting to delete group : " + exc.message)
+
+            message = 'Deleted group successfully.'
+            has_changed = True
+
+    return group_id, message, has_changed
+
+
+def retrieve_group_instances(client, module, group_id):
+    wait_timeout = module.params.get('wait_timeout')
+    wait_for_instances = module.params.get('wait_for_instances')
+
+    if wait_timeout is None:
+        wait_timeout = 300
+
+    wait_timeout = time.time() + wait_timeout
+    target = module.params.get('target')
+    state = module.params.get('state')
+    instances = list()
+
+    if state == 'present' and group_id is not None and wait_for_instances is True:
+
+        is_amount_fulfilled = False
+        while is_amount_fulfilled is False and wait_timeout > time.time():
+            amount_of_fulfilled_instances = 0
+            active_instances = client.get_elastigroup_active_instances(group_id=group_id)
+
+            for active_instance in active_instances:
+                if active_instance.get('private_ip') is not None:
+                    amount_of_fulfilled_instances += 1
+                    instances.append(active_instance)
+
+            if amount_of_fulfilled_instances >= target:
+                is_amount_fulfilled = True
+
+            time.sleep(10)
+
+    return instances
+
+
+def find_group_with_same_name(groups, name):
+    group_found = False
+    group_id = ""
+    for group in groups:
+        if group['name'] == name:
+            group_found = True
+            group_id = group.get('id')
+            break
```

Comment:
I think these 3 lines can be replaced with `return True, group.get('id')` and the below return can change to `return False, ''`

Reply:
Very elegant! Made this change as well.